src/encoded/tests/test_permissions.py

```diff
@@ -8,59 +8,142 @@ def submission_center_user_app(testapp, test_submission_center, smaht_gcc_user):
+        'filename': 'my.fastq.gz',
+        'status': 'shared',    # this status is important as this will make it viewable by consortium
+        'consortiums': [
+            test_consortium['uuid']
+        ]
+    }
+    res = testapp.post_json('/FileSubmitted', item)
     return res.json['@graph'][0]
 
 
-class TestSubmissionCenterPermissions:
+class TestPermissionsHelper:
+
+    @staticmethod
+    def validate_get_permissions(*, restricted_app, restricted_expected_status,
+                                  admin_app, admin_expected_status,
+                                  anon_app, anon_expected_status,
+                                  item_uuid):
+        """ Tests a series of app/status combinations for GET based on the callers desired behavior """
+        restricted_app.get(f'/{item_uuid}', status=restricted_expected_status)
+        admin_app.get(f'/{item_uuid}', status=admin_expected_status)
+        anon_app.get(f'/{item_uuid}', status=anon_expected_status)
+
+
+class TestSubmissionCenterPermissions(TestPermissionsHelper):
     """ Tests permissions scheme centered around the submission center ie: can they view associated items,
-        can they create/edit them etc
+        can they create/edit them etc - all testing with default status
     """
 
-    @staticmethod
-    def test_submission_center_file_permissions_view(submission_center_user_app, smaht_gcc_user, testapp,
-                                                     anontestapp, file):
+    def test_submission_center_file_permissions_view(self, submission_center_user_app, smaht_gcc_user, testapp,
+                                                     anontestapp, submission_center_file):
         """ Tests that a user associated with a submission center can view an uploaded permissioned
             file, an anonymous user cannot and an admin user can """
-        submission_center_user_app.get(f'/{file["uuid"]}', status=200)
-        anontestapp.get(f'/{file["uuid"]}', status=403)
-        testapp.get(f'/{file["uuid"]}', status=200)
+        uuid = submission_center_file["uuid"]
+        self.validate_get_permissions(
+            restricted_app=submission_center_user_app, restricted_expected_status=200,
+            admin_app=testapp, admin_expected_status=200,
+            anon_app=anontestapp, anon_expected_status=403,
+            item_uuid=uuid
+        )
 
-        # patch the file status so it has no submission_center, user should now no longer see
-        testapp.patch_json(f'/{file["uuid"]}?delete_fields=submission_centers', {}, status=200)
-        submission_center_user_app.get(f'/{file["uuid"]}', status=403)
-        anontestapp.get(f'/{file["uuid"]}', status=403)
-        testapp.get(f'/{file["uuid"]}', status=200)
+        # patch the file status, so it has no submission_center therefore user should no longer see
+        testapp.patch_json(f'/{submission_center_file["uuid"]}?delete_fields=submission_centers', {}, status=200)
```

Comment:
why patch and not just use a different submission center? won't it be faster and possibly more robust? what mechansim puts it back?

Reply:
All datafixtures are function scoped so the changes in the test are wiped before the next test. Even with a small # of tests here they seem to go very fast per test even when building multiple data model elements so have preferred this over the workbook in the meantime.